### PR TITLE
CMake and unit tests Refcactoring

### PR DESCRIPTION
**Pull request by nidegen**
I slightly adapted cmake and the tests structure

### CODE REVIEW

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+################################
+# OpenMapper Tests
+################################
+
+SET(OM_WRAPPER_TESTS "OpenMapperTests")
+# Add test src files
+ADD_EXECUTABLE(${OM_WRAPPER_TESTS}
+  ${CMAKE_CURRENT_LIST_DIR}/main.cc
+  ${CMAKE_CURRENT_LIST_DIR}/pose_test.cc)
```
Comment (gocarlos) [inline test/CMakeLists.txt:17]: Is not runnable right?

File: test/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+# Link test executable against gtest & gtest_main
+TARGET_LINK_LIBRARIES(${OM_WRAPPER_TESTS}
+  gtest gtest_main
+debug ${OM_WRAPPER_LIB} optimized ${OM_WRAPPER_LIB})
+
+ADD_TEST(${OM_WRAPPER_TESTS} ${OM_WRAPPER_TESTS})
+
+TARGET_INCLUDE_DIRECTORIES(${OM_WRAPPER_TESTS} PUBLIC include)
+
+SET_TARGET_PROPERTIES(${OM_WRAPPER_TESTS} PROPERTIES RUNTIME_OUTPUT_DIRECTORY ../bin/ )
```
Comment (gocarlos) [inline test/CMakeLists.txt:28]: I'd also add an entry to the .gitignore i.o. to not have the binaries on github

File: test/main.cc [hunk 3]
```diff
@@ -0,0 +1,6 @@
+#include "gtest/gtest.h"
```
Comment (gocarlos) [inline test/main.cc:1]: Are you sure that the tests are run, I mean will "pose_test" run?

File: CMakeLists.txt [hunk 4]
```diff
@@ -1,46 +1,88 @@
...
+  ARCHIVE_OUTPUT_DIRECTORY_DEBUG ../lib/
+  ARCHIVE_OUTPUT_DIRECTORY_RELWITHDEBINFO ../lib/
+  ARCHIVE_OUTPUT_DIRECTORY_RELEASE ../lib/)
+
+
+################################
+# Dependencies
+################################
+
+# # OpenCV
```
Comment (gocarlos) [inline CMakeLists.txt]: I'd add here a TODO(nidegen) or some comment on why this is commented out. 
Best practice are both :)

File: CMakeLists.txt [hunk 5]
```diff
@@ -1,46 +1,88 @@
...
 
-include_directories(include)
+PROJECT(OpenMapper)
+SET(OM_WRAPPER_LIB "OpenMapper")
+SET(OM_WRAPPER_APP "OpenMapperApp")
 
-##############
-# External Library
-##############
+SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
```
Comment (gocarlos) [inline CMakeLists.txt]: for cmake version >=3.1 the recommended flag is: 
set(CMAKE_CXX_STANDARD 11) 
set(CMAKE_CXX_STANDARD_REQUIRED ON) 

ref: 
https://stackoverflow.com/questions/37621342/cmake-will-not-compile-to-c-11-standard
https://cmake.org/cmake/help/v3.1/manual/cmake-compile-features.7.html

**Review by gocarlos — CHANGES_REQUESTED** [review]
Very nice job on the cmakelists & gtests. 
I've made some comments, which you could address before merging. 
Other than that LGTM

File: test/CMakeLists.txt (hunk 2 above)
Comment (nidegen) [reply test/CMakeLists.txt:28]: oops, yes of course, my mistake

File: test/CMakeLists.txt (hunk 1 above)
Comment (nidegen) [reply test/CMakeLists.txt:17]: it is

File: test/main.cc (hunk 3 above)
Comment (nidegen) [reply test/main.cc:1]: yes google test handle this all

File: CMakeLists.txt (hunk 5 above)
Comment (nidegen) [reply CMakeLists.txt]: ok

**Comment by nidegen** [comment]
thansk for the review!

**Review by nidegen — COMMENTED** [review]
i have adressed the comments

**Comment by nidegen** [comment]
not sure if i correctly proceeded to marl all review requests as done on github, but i just merged:)

**Pull request merged**